### PR TITLE
Feature/error to json

### LINKED ISSUE #63: Implement `error-to-json`
See [utils-error-to-json](https://github.com/kgryte/utils-error-to-json). `@stdlib/utils/error-to-json`.

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves #63 .

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

* [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
* [x] Read and understand the [Code of Conduct][code-of-conduct].
* [x] Read and understood the [licensing terms][license].
* [x] Searched for existing issues and pull requests __before__ submitting this pull request.
* [x] Filed an issue (or an issue already existed) __prior to__ submitting this pull request.
* [x] Rebased onto latest `develop`.
* [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

* Contains a straight-forward port of [utils-error-to-json](https://github.com/kgryte/utils-error-to-json) 

## Related Issues

> Does this pull request have any related issues?

This pull request:

* resolves #
* fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

---

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/master/CONTRIBUTING.md
[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/master/CODE_OF_CONDUCT.md
[license]: https://github.com/stdlib-js/stdlib/blob/master/LICENSE

### CODE REVIEW

**Comment by kgryte** [comment]
@Planeshifter Can you rebase this on the latest commit? I pushed code which has a broken `require` which just got fixed.

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: Fix parameter.

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: Can we end each property description with a period? This is our common practice for `options`.

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/`function`/function

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: Can we remove the `console.log` statements and replace with

``` javascript
var obj = toJSON( err );
/* returns

*/
```

File: lib/node_modules/@stdlib/utils/error-to-json/examples/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/examples/index.js]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/examples/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/examples/index.js]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/examples/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/examples/index.js]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/examples/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/examples/index.js]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/examples/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/examples/index.js]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/examples/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/examples/index.js]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/examples/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/examples/index.js]: s/\<stack\>/.../

File: lib/node_modules/@stdlib/utils/error-to-json/lib/ctors.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/lib/ctors.js]: Either remove the heading or replace with `MAIN`,

File: lib/node_modules/@stdlib/utils/error-to-json/lib/error_to_json.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/lib/error_to_json.js]: `require( 'object-keys' ).shim();`

File: lib/node_modules/@stdlib/utils/error-to-json/lib/type.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/lib/type.js:22]: We should probably add a note in the README that determining the error type only works if provided an `error` from the same realm. Otherwise, if we want this pkg to work cross-realm, we need to use `@stdlib/utils/type-of` or `@stdlib/utils/constructor-name`, which rely on either `Object.prototype.toString` (which can be faked) or the constructor `name`, which can be set.

File: lib/node_modules/@stdlib/utils/error-to-json/test/fixtures/customerror.subclass.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/fixtures/customerror.subclass.js]: Add a TODO here

```
/* jshint evil:true */ // TODO: replace with eslint
```

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.ctors.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.ctors.js]: Use `@stdlib/utils/is-array`.

Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.ctors.js]: Add

``` javascript
t.ok( true, __filename );
```

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.ctors.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.ctors.js]: s/test/tape/

Affects subsequent usage below.

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]: s/test/tape/

Affects subsequent usage below.

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]:
``` javascript
t.ok( true, __filename );
t.strictEqual( typeof toJSON, 'function', '...' );
```

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]: Use `@stdlib/utils/is-plain-object`.

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]: s/undefined/void 0/

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]: s/undefined/void 0/

``` javascript
t.strictEqual( json.stack, void 0, '...' );
```

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]: s/undefined/void 0/

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]: s/undefined/void 0/

``` javascript
t.strictEqual( json.name, void 0, '...' );
```

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.js]: Use `opts.skip`. Can set `opts.skip` at the top of file like we do for other packages.

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.type.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.type.js]:
``` javascript
t.ok( true, __filename );
t.strictEqual( typeof typeName, 'function', '...' );
```

File: lib/node_modules/@stdlib/utils/error-to-json/test/test.type.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/test/test.type.js]: s/test/tape/

Affects subsequent usage below.

**Review by kgryte — CHANGES_REQUESTED** [review]

**Comment by Planeshifter** [comment]
Rebased and ready for final review.

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: A bit of an awkward title. Sense a better title would be to match the function name.

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: Options:

```
toJSON
errorToJSON
toJSONError
err2json
error2json
errToJSON
```
Ideally, we would have a name which more or less corresponds to what will be exported to namespace.

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/error-to-json/README.md]: > The function throws an error for cross-realm error objects.

Comment (kgryte) [reply lib/node_modules/@stdlib/utils/error-to-json/README.md]: > The function uses `instanceof` to infer the error type.

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/utils/error-to-json/README.md
Comment (Planeshifter) [reply lib/node_modules/@stdlib/utils/error-to-json/README.md]: Yes, I was thinking about that. All are kind of ugly. I am inclined to go with `err2json`. Your stance?

Comment (kgryte) [reply lib/node_modules/@stdlib/utils/error-to-json/README.md]: `err2json` would probably be mine. I am okay with using `2` as a bridge when it makes sense. In other cases, the "ugly" way is the only way; e.g., `float64ToFloat32`.

**Comment by Planeshifter** [comment]
Changed the module name and updated the note. Thanks for reviewing!

File: lib/node_modules/@stdlib/utils/lib/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/lib/index.js]: The module path is incorrect.

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/utils/lib/index.js
Comment (Planeshifter) [reply lib/node_modules/@stdlib/utils/lib/index.js]: Thanks for catching this!

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
@Planeshifter We should at least wait until Circle CI has passed for the latest commit before merging.

**Comment by Planeshifter** [comment]
Roger, sorry for my overzealousness.

**Pull request merged**
